### PR TITLE
BasicAuthExtractor should work on Authorization: Basic by default

### PR DESCRIPTION
**Pull request by leleuj**

### CODE REVIEW

**Comment by mmoayyed** [comment]
:+1

**Comment by mmoayyed** [comment]
We should also clean up the CAS basic authn integration later

**Comment by leleuj** [comment]
Can you elaborate a little more?

**Comment by mmoayyed** [comment]
Sure.

CAS does this:

```java
 private static final String AUTHZ_HEADER = "Authorization";
...
final BasicAuthExtractor extractor = new BasicAuthExtractor(AUTHZ_HEADER, "Basic", this.getClass().getSimpleName());
```

We should use:

```java
final BasicAuthExtractor extractor = new BasicAuthExtractor(this.getClass().getSimpleName());
```

File: pac4j-core/src/main/java/org/pac4j/core/context/HttpConstants.java [hunk 1]
```diff
@@ -42,6 +42,8 @@
 
     String AUTHORIZATION_HEADER = "Authorization";
 
+    String BASIC_HEADER_PREFIX = "Basic ";
```
Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/context/HttpConstants.java]: Do we need to check for case? BASIC? basic? Is "Basic" pretty much the standard?

Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/context/HttpConstants.java]: According to https://tools.ietf.org/html/rfc2617, it's "Basic ", but I think we could use a `equalsIgnoreCase` in the `BasicAuthExtractor`. No one has ever reported an issue on this...

Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/context/HttpConstants.java]: Sounds good.

Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/context/HttpConstants.java]: Let's get this merged. Looks pretty done. I ll work on the CAS side of things based on the snapshot here.

**Pull request merged**